feat: Add post endpoint

rafaeelaudibert opened the pull request:
This endpoint is used for our public endpoint, where we'll be able to create short URLs from our public page

jose-sequeira commented on rust/links/src/redirect/redirect_service.rs:
```diff
@@ -77,6 +85,9 @@ impl InternalRedirectService {
 }
 
+const TWENTY_FOUR_HOURS_IN_SECONDS: u64 = 60 * 60 * 24;
+const PHOG_GG_DOMAIN: &str = "phog.gg";
```
Should we have this on the configuration instead of hardcoded?

jose-sequeira replied:
in `config.rs` we already read from environment set config, you can just add a new variable

rafaeelaudibert replied:
Nice, thanks for the lead

rafaeelaudibert commented on rust/links/src/middleware/rate_limit.rs:
```diff
@@ -0,0 +1,89 @@
+                .get::<ClientIp>()
+                .map(|ip| ip.0.to_string())
+        })
+        .unwrap_or_else(|| "unknown".to_string());
```
Just if no IP is shared, I'm fine with that

rafaeelaudibert commented on rust/links/src/redirect/redirect_service.rs:
```diff
@@ -97,6 +108,30 @@ impl RedirectServiceTrait for ExternalRedirectService {
+
+    async fn store_url(
+        &self,
+        destination: &str,
+        short_string: &str,
```
That's ok, the onus is on the user, this is a free service

rafaeelaudibert commented on rust/links/src/middleware/rate_limit.rs:
```diff
@@ -0,0 +1,89 @@
+    S::Future: Send + 'static,
+{
+    // Get the client IP from the request
+    let ip = req
+        .headers()
+        .get("x-forwarded-for")
+        .and_then(|h| h.to_str().ok())
+        .map(String::from)
+        .or_else(|| {
+            req.extensions()
+                .get::<ClientIp>()
+                .map(|ip| ip.0.to_string())
+        })
+        .unwrap_or_else(|| "unknown".to_string());
```
Maybe? @jose-sequeira would you agree here? This is very naive anyway, so I should probably just do this

jose-sequeira replied:
Yes, this is reasonable!

jose-sequeira commented on rust/links/src/router.rs:
```diff
@@ -35,15 +39,27 @@ where
 
-    let links_router = Router::new()
+    // Create rate limiter for POST /ph endpoint
+    let rate_limiter = create_rate_limiter();
```
Should rate limiter be at the Proxy level instead of service level?

meikelmosby replied:
i'll put it into my todos but lets not block this right now bc of this

meikelmosby replied:
talked to michi and it seems we do not really need to care about rate limiting here as we get a circuit breaker for free so... ditch it

rafaeelaudibert replied:
I love our Infra team

rafaeelaudibert committed:
feat: Add post endpoint

This endpoint is used for our public endpoint, where we'll be able to create short URLs from our public page

rafaeelaudibert committed:
refactor: Store `phog.gg` domain as config

This allows us to more easily modify this for eu/us via an env variable

rafaeelaudibert committed:
feat: Return formatted URL

meikelmosby commented on rust/links/src/router.rs:
```diff
@@ -30,20 +35,26 @@ where
-    let links_router = Router::new()
-        .route("/pr/:origin_key", get(external_redirect_url))
-        .route("/pr/:origin_key/", get(external_redirect_url))
+    let links_public_router = Router::new()
```
lets call this links_external_router to be consistent with the namings?

meikelmosby commented on rust/links/src/router.rs:
```diff
@@ -30,20 +35,26 @@ where
+        .route("/ph", post(external_store_url))
+        .route("/ph/", post(external_store_url));
+
+    let links_private_router = Router::new()
```
lets call this links_internal_router to be consistent with the namings?

rafaeelaudibert committed:
ci: Fix CI stuff

rafaeelaudibert committed:
lint: Fix lint

frankh committed:
feat: add log capture service (#32229)

Co-authored-by: Oliver Browne <oliver@posthog.com>

frankh committed:
fix: fix log-capture package name again (#32240)

rafaeelaudibert committed:
feat: Add CORS

Copied over from flags

rafaeelaudibert committed:
fix: Fix lint by removing shutdown

rafaeelaudibert committed:
refactor: Reintroduce shutdown

rafaeelaudibert committed:
fix: Fix lint
